### PR TITLE
Make ErgoLikeTransaction.id a tagged String (unify with Scorex)

### LINKED ISSUE #216: Unify ErgoLikeTransaction.id between Ergo, Scorex and Sigma
Related to #205 and Scorex [PR 294](https://github.com/ScorexFoundation/Scorex/pull/294)

Scorex `Transaction.id` is of type `ModifierId` which is tagged `String`.
Sigma `ErgoLikeTransaction.id` is of type `Array[Byte]`.

This prevents mixing of this two traits.

Todo:
- [x] backport to `hotfix/v0.9.5` branch;
- [x] publish v0.9.5;

### PR DESCRIPTION
**Pull request by greenhat**
Ref #216 

Todo:
- [x] use `ModifierId` from `scorex-util`;
- [x] wait https://github.com/ScorexFoundation/scorex-util/pull/6 to be merged;
- [x] wait for `scorex-util` v1.1.0 is published;
- [x] switch `scorex-util` version to public v1.1.0;
- [x] wait for `scrypto` v2.1.3 is published;
- [x] switch `scrypto` version to public v2.1.3;

### CODE REVIEW

**Comment by greenhat** [comment]
@ergomorphic I suggest we merge this workaround and after  
https://github.com/ScorexFoundation/Scorex/pull/294 is merged come back to #216 (leave it open) for a proper unification fix if necessary.

**Comment by greenhat** [comment]
@ergomorphic Check out my take on making tx id a tagged String.

File: src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala [hunk 1]
```diff
@@ -69,20 +67,19 @@ class ErgoLikeTransaction(override val inputs: IndexedSeq[Input],
...
 
-  override type IdType = Digest32
-
-  override lazy val id: IdType = Blake2b256.hash(messageToSign)
+  override lazy val id: ModifierId = Blake2b256.hash(messageToSign).toModifierId
 
   override def equals(obj: Any): Boolean = obj match {
-    case tx: ErgoLikeTransaction => this.id sameElements tx.id  //we're ignoring spending proofs here
+    //we're ignoring spending proofs here
+    case tx: ErgoLikeTransaction => this.id == tx.id
```
Comment (ergomorphic) [inline src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala]: @greenhat if we override equals we should also override hashCode. I understand that is was like this, but please fix it.

**Comment by ergomorphic** [comment]
LGTM, @catena2w  we are waiting for your review.

File: src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala (hunk 1 above)
Comment (greenhat) [reply src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala]: @ergomorphic I should've spotted this myself. Good catch! Thanks!

File: src/main/scala/org/ergoplatform/ErgoBox.scala [hunk 2]
```diff
@@ -133,23 +134,25 @@ object ErgoBox {
             proposition: Value[SBoolean.type],
             additionalTokens: Seq[(TokenId, Long)] = Seq(),
             additionalRegisters: Map[NonMandatoryRegisterId, _ <: EvaluatedValue[_ <: SType]] = Map(),
-            transactionId: Array[Byte] = Array.fill(32)(0: Byte),
+            transactionId: ModifierId = ModifierId @@ Array.fill[Byte](32)(0.toByte).toModifierId,
```
Comment (catena2w) [inline src/main/scala/org/ergoplatform/ErgoBox.scala]: Why `ModifierId @@`  here?

Comment (greenhat) [reply src/main/scala/org/ergoplatform/ErgoBox.scala]: @catena2w Leftovers. Good catch! Thanks! Fixed.

**Pull request merged**